### PR TITLE
Potentially trustworthy origins

### PR DESCRIPTION
**Pull request by ewilligers**
The manifest and action urls must be "potentially trustworthy"
e.g. https or 127.0.0.1

We may not need to refer to "secure contexts" - see #27

***
<a href="https://pr-preview.s3.amazonaws.com/ewilligers/web-share-target/pull/69.html" title="Last updated on Aug 30, 2018, 1:36 PM GMT (42a39ba)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/web-share-target/69/3fccfd1...ewilligers:42a39ba.html" title="Last updated on Aug 30, 2018, 1:36 PM GMT (42a39ba)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -303,6 +303,11 @@ <h3>
           <li>If <var>share target</var> is <code>undefined</code>, then return
           <code>undefined</code>.
           </li>
+          <li>If the <a data-cite="!URL#concept-url-origin">origin</a> of <var>
```
Comment (mgiuca) [inline index.html]: I'm not sure we need to check that the manifest itself is PT. Seems outside the scope of WST to care whether the manifest itself is PT, just that the site should be trustworthy. @dominickng what do you think?

File: index.html [hunk 2]
```diff
@@ -303,6 +303,11 @@ <h3>
           <li>If <var>share target</var> is <code>undefined</code>, then return
```
Comment (mgiuca) [inline index.html]: (Above)

I think you should remove the "in a secure context" language and link to issue #27. I think the below is sufficient normative text.

Though you might want to add some non-normative text (in an example section) that says it needs to be secure.

File: index.html (hunk 1 above)
Comment (dominickng) [reply index.html]: Hmmm. Aren't data URLs considered as opaque origins, and therefore "Not trustworthy"? I think it's a legitimate use case to have a manifest URL be a data URL, so this might actually break some sites that use that.

I think I agree that if you want to keep the potentially trustworthy check, it should be the destination of the web share target, rather than the manifest.

Comment (mgiuca) [reply index.html]: Discussed with Eric yesterday. I think it's a reasonable argument that the manifest has to be secure, otherwise a MITM can just inject any web share target without the site's knowledge.

Yes, it's true that data URLs have an opaque origin and are therefore "not trustworthy". That seems bad (in general). I don't feel like a hack here to allow data URLs is the right place. How does the PWA check deal with these data URLs?

Comment (mgiuca) [reply index.html]: According to this: https://www.w3.org/TR/secure-contexts/#is-url-trustworthy

"Note: This aligns the definition of a secure context with the de facto "data: URL as opaque origin" behavior that a majority of today’s browsers have agreed upon, rather than the de jure "data: URL inherits origin" behavior defined in HTML."

What it's saying is: yeah data URLs should inherit trustworthiness from the containing document, but for shitty historical reasons they are not trusted. I feel like we should just follow suit and not trust them.

Comment (dominickng) [reply index.html]: I don't think this is the right level to be protecting against MITM attacks. That's the job of TLS, and besides, MITM gives you significantly more issues than a web share target injection.

The PWA check doesn't do this potentially untrustworthy check. In fact, in Chrome, content::IsOriginSecure actually returns `true` for data:// URLs (https://cs.chromium.org/chromium/src/content/common/origin_util.cc?type=cs&q=content/common/origin_util.cc&sq=package:chromium&g=0&l=42 - trace into url::GetSecureSchemes())

File: index.html [hunk 3]
```diff
@@ -662,10 +667,11 @@ <h2>
         from an online index, rather than a set of targets that the end user
         has explicitly installed or registered.
         </li>
-        <li>The requirement that the web share target be a <a data-cite=
-        "SECURE-CONTEXTS#secure-context">secure context</a> is to prevent
-        private user data from being transmitted to a party that does not
-        control the origin in question, or in clear text over the network.
+        <li>The requirement that the web share target's origin be
+          <a data-cite="!SECURE-CONTEXTS/#is-origin-trustworthy">potentially
```
Comment (mgiuca) [inline index.html]: Nit: Drop the '!' (which makes this a normative requirement).
